src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -226,24 +226,21 @@ private static void CheckEndReachable(object startSegment, object endSegment)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private long GetLength(in SequencePosition start, in SequencePosition end)
         {
-            GetTypeAndIndices(start.GetInteger(), end.GetInteger(), out SequenceType type, out int startIndex, out int endIndex);
-
+            int startIndex = GetIndex(start);
+            int endIndex = GetIndex(end);
             object startObject = start.GetObject();
             object endObject = end.GetObject();
-            if (type == SequenceType.MultiSegment && startObject != endObject)
+            if (startObject != endObject)
```

Comment:
If either startObject or endObject are null, this method will throw a NullReferenceException. @pakrym, what should be the correct behavior?

My suggestion would be to add a null check up front (just like https://github.com/dotnet/corefx/pull/29135/files#diff-6a3b2bb4fe535f3ac47622b764877747R321)
```C#
if (startObject == null || endObject == null) return 0;
```

Reply:
We always call them only on _start/_end so they might be null only in default case if all our constructor checks are correct. But in default case they would both be null.

Reply:
> If either startObject or endObject are null, this method will throw a NullReferenceException (when calling endSegment.RunningIndex, for example). @pakrym, what should be the correct behavior?

You're wrong. endSegment.RunningIndex is not used if either startObject or endObject are null.

Reply:
> My suggestion would be to add a null check up front

It is makes this method slower. It is works correct on default sequence.

Reply:
> We always call them only on _start/_end so they might be null only in default case if all our constructor checks are correct.

I don't understand. Both start and end are passed in by the user. If they pass in start as null and end as not null, we get an exception. Where are we making sure that both are null or both are not null?

> You're wrong. endSegment.RunningIndex is not used if either startObject or endObject are null.

Debug.Asserts do not fire in Release. If both are null, we don't go down this path. But if only one is null, we end up casting it to ReadOnlySequenceSegment and calling RunningIndex.

Reply:
>  don't understand. Both start and end are passed in by the user. If they pass in start as null and end as not null, we get an exception. Where are we making sure that both are null or both are not null?

I just realized (thanks @pakrym), that this is a private method only ever called by the Length property with _sequenceStart and _sequenceEnd only. We should remove the parameters from this method all together.

Reply:
> But if only one is null, we end up casting it to ReadOnlySequenceSegment and calling RunningIndex.

Yes but such structure is broken. Why we should protect this method from broken structure? Many other methods have the same issue on broken structure.

Reply:
As I understand we only protect ROS from breaking type safety. But not from all variants of broken structure.

Reply:
> Yes but such structure is broken. Why we should protect this method from broken structure? Many other methods have the same issue on broken structure.

> As I understand we only protect ROS from breaking type safety. But not from all variants of broken structure.

Yes, you are right. This change is fine as is. Thanks.